Framework/Tools/Emulator/Ssl.cs

```diff
@@ -608,9 +650,13 @@ public int DataAvailable( int socket )
 
             if (!GetSslData(socket, out ssd)) return (int)SocketError.SocketError;
 
-            if(!_socketsDriver.GetSocket(socket, out sock)) return (int)SocketError.SocketError;
+            if (!_socketsDriver.GetSocket(socket, out sock)) return (int)SocketError.SocketError;
 
-            return sock.Poll(0,SelectMode.SelectRead) ? 1024 : 0;
+            var readData = ssd.m_readData;
+            int ret = readData != null ?

```

Comment:
Done too!
I agree with you 100%. 
Remember that I'm not the author of these changes, I just applied the patch coming from the AMQP Lite project.